website/content/en/highlights/2022-05-24-vrl-encryption.md

```diff
@@ -0,0 +1,102 @@
+---
+date: "2022-05-24"
+title: "New encrytion functions for VRL"
+description: "Encrypt and decrypt data in VRL"
+authors: ["jszwedko"]
+pr_numbers: []
+release: "0.22.0"
+hide_on_release_notes: false
+badges:
+  type: enhancement
+---
+
+VRL now contains facilicities for for encrypting and decrypting field values using [AES][AES] encryption with user
+provided keys via new [`encrypt`][encrypt] and [`decrypt`][decrypt] functions. A [`random_bytes`][random_bytes] function
+was added to make it easy to generate initialization vectors for the [`encrypt`][encrypt] function.
+
+These can be useful to:
+
+- Decrypt encrypted data that is sent to Vector for processing
+- Encrypting sensitive fields in Vector for storage in an external service that can be decrypted on-demand if the
+  original value is required
+
+## Encrypting data
+
+To encrypt data in VRL with an encryption key you can provide an encryption key. Typically the key is injected via an
+environment variable.
+
+An example with input:
+
+```json
+{ "plaintext": "super secret message" }
+```
+
+```text
```

Comment:
Not coffeescript?

Reply:
I thought about it, but when I did it in `vim` the highlighting looked poor; primarily due to the semicolons.

Reply:
No worries - was wondering if the highlighting was weird with the "non-code lines"

Reply:
🤔 the website handles it better than `vim` was:

<img width="978" alt="Screen Shot 2022-05-24 at 3 02 51 PM" src="https://user-images.githubusercontent.com/316880/170139582-c8d166e3-999d-43e1-a37f-801eab0cac66.png">

The main downside appears to that VRL's comments are different than Coffeescript's.

Maybe with the new site we can add a syntax highlighter specifically for VRL.